Bond/Extensions/OSX/NSTableView+Bond.swift

```diff
@@ -0,0 +1,138 @@
+  private var delegate: BNDTableViewDelegate?
+
+  private init(array: ObservableArray<T>, tableView: NSTableView, delegate: BNDTableViewDelegate) {
+    self.tableView = tableView
+    self.delegate = delegate
+    self.array = array
+    super.init()
+
+    tableView.setDataSource(self)
+    tableView.setDelegate(self)
+    tableView.reloadData()
+
+    array.observeNew { [weak self] arrayEvent in
+      guard let unwrappedSelf = self,
+                tableView = unwrappedSelf.tableView else { return }
+
+      switch arrayEvent.operation {
+      case .Batch(let operations):
+        tableView.beginUpdates()
+        for diff in changeSetsFromBatchOperations(operations) {
+          BNDTableViewDataSource.applyRowUnitChangeSet(diff, tableView: tableView)
+        }
+        tableView.endUpdates()
+      case .Reset:
+        tableView.reloadData()
+      default:
+        tableView.beginUpdates()
+        BNDTableViewDataSource.applyRowUnitChangeSet(arrayEvent.operation.changeSet(), tableView: tableView)
+        tableView.endUpdates()
+      }
+    }.disposeIn(bnd_bag)
+  }
+
+  private class func applyRowUnitChangeSet(changeSet: ObservableArrayEventChangeSet, tableView: NSTableView) {
+    switch changeSet {
+    case .Inserts(let indices):
+      // FIXME: How to use .Automatic effect a-la UIKit?
+      tableView.insertRowsAtIndexes(NSIndexSet(set: indices), withAnimation: .EffectNone)
+    case .Updates(let indices):
+      tableView.reloadDataForRowIndexes(NSIndexSet(set: indices), columnIndexes: NSIndexSet())
+    case .Deletes(let indices):
+      tableView.removeRowsAtIndexes(NSIndexSet(set: indices), withAnimation: .EffectNone)
+    }
+  }
+
+  /// MARK - NSTableViewDataSource
+
+  @objc func numberOfRowsInTableView(tableView: NSTableView) -> Int {
+    return array.count
+  }
+
+  /// MARK - NSTableViewDelegate
+
+  @objc func tableView(tableView: NSTableView, viewForTableColumn tableColumn: NSTableColumn?, row: Int) -> NSView? {
+    return delegate?.createCell(row, array: array, tableView: tableView)
+  }
+
+  override func forwardingTargetForSelector(aSelector: Selector) -> AnyObject? {
+    guard let delegate = delegate as? AnyObject
+    where delegate.respondsToSelector(aSelector) else {
+      return self
+    }
+
+    return delegate
+  }
+
+  override func respondsToSelector(aSelector: Selector) -> Bool {
+    guard let delegate = delegate as? AnyObject
+    where delegate.respondsToSelector(aSelector) else {
+      return super.respondsToSelector(aSelector)
+    }
+
+    return true
+  }
+}
+
+extension NSTableView {
+  private struct AssociatedKeys {
+    static var BondDataSourceKey = "bnd_BondDataSourceKey"
+  }
+}
+
+public extension ObservableType where EventType: ObservableArrayEventType {
+
+  private typealias ElementType = EventType.ObservableArrayEventSequenceType.Generator.Element
+
+  public func bindTo(tableView: NSTableView, delegate: BNDTableViewDelegate) -> DisposableType {
```

Comment:
I didn't know that difference. Let's leave it as it is then. For UIKit I had to create "proxy" delegates to enabled hooks for methods not implemented by Bond.